src/of_type.jl

```diff
@@ -4,24 +4,113 @@ using Random: randexp
 # Core Type Definitions
 # ========================================================================
 
-# Abstract base type for all Of types
+"""
+    OfType
+
+Abstract base type for all types in the `of` type system.
+
+The `of` type system provides a declarative way to specify parameter types for 
+probabilistic programming. All `of` types encode their specifications (dimensions, 
+bounds, etc.) in type parameters, allowing them to be used as actual Julia types 
+in type annotations.
+
+# Subtypes
+- `OfReal{T,Lower,Upper}`: Bounded or unbounded floating-point numbers
+- `OfInt{Lower,Upper}`: Bounded or unbounded integers  
+- `OfArray{T,N,Dims}`: Arrays with specified element type and dimensions
+- `OfNamedTuple{Names,Types}`: Named tuples with typed fields
+- `OfConstantWrapper{T}`: Wrapper marking a type as constant/hyperparameter
+
+# See also
+[`of`](@ref), [`@of`](@ref)
+"""
 abstract type OfType end
 
-# Wrapper type for symbolic references in bounds
+"""
+    SymbolicRef{S}
+
+Wrapper type for symbolic references in bounds and dimensions.
+
+Used internally to encode references to other fields when specifying bounds or dimensions 
+that depend on constant fields. For example, when using `@of(n=of(Int; constant=true), 
```

Comment:
Bounds or dimensions may also depend on other parameters. 
```suggestion
that depend on constant fields or other parameters. For example, when using `@of(n=of(Int; constant=true), 
```